plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java

```diff
@@ -884,4 +925,19 @@ protected SqlExecutor onRemoteDatabase()
     {
         return clickhouseServer::execute;
     }
+
+    private static Function<ZoneId, String> clickhouseDateTimeInputTypeFactory(String inputTypePrefix)
+    {
+        return zone -> format("%s('%s')", inputTypePrefix, zone);
+    }
+
+    private static BiFunction<LocalDateTime, ZoneId, String> clickhouseDateTimeInputLiteralFactory()
+    {
+        return (inputLiteral, zone) -> format("'%s'", DateTimeFormatter.ofPattern("uuuu-MM-dd HH:mm:ss").format(inputLiteral));
+    }
+
+    private static BiFunction<LocalDateTime, ZoneId, String> trinoTimestampExpectedLiteralFactory()
+    {
+        return (expectedLiteral, zone) -> format("TIMESTAMP '%s'", DateTimeFormatter.ofPattern("uuuu-MM-dd HH:mm:ss VV").format(expectedLiteral.atZone(zone).withFixedOffsetZone()));
+    }
```

Comment:
useful idea to be able to re-use literals. But personally I think inlined strings are easier to read when looking at the tests and they also prevent the possibility of something passing because the literal factory applied some changes.

Reply:
i'd extract the `DateTimeFormatter.ofPattern("uuuu-MM-dd HH:mm:ss VV")` out of lambda, but still within this method

for tests, it doesn't matter much though